src/camera.c

```diff
@@ -159,7 +219,8 @@ open_pipewire_camera_remote (const char *app_id,
    * permission store to set up permissions.
    */
   permission_items[0] = PW_PERMISSION_INIT (PW_ID_CORE, PW_PERM_RWX);
-  permission_items[1] = PW_PERMISSION_INIT (PW_ID_ANY, 0);
+  permission_items[1] = PW_PERMISSION_INIT (remote->node_factory_id, PW_PERM_R);
```

Comment:
This is handled by `module-portal` in PipeWire. It was part of the PR as the `module-portal` had not been brought over to 0.3.

Reply:
Mockups are just that: sketches that present an rough idea at a certain point in time. The ones we have for the camera portal are more than 3 years old now. I would not put too much weight on them.

> But for applications to see what cameras it has access to, it should just use PipeWire directly.

The point here was to give the user control over what cameras the application has access to. I don't see how you solve that without somebody enumerating cameras with in a user-presentable way.

And even if you give up on access control and say: let the app talk to pipewire directly, you just drop the "figure out user-presentable camera enumeration" problem onto the application.

Reply:
> Pipewire still queries the permission store though, so whatever the user selects doesn't affect what cameras the application will get access to; it seems more like a outsourcing a piece of UI for selecting a camera, without any security benefits.

I was never entirely happy with this approach, and the recent "oops cameras don't work with pw 0.3" shows why: We treat pipewire as this magic black box. But it doesn't actually work, and we loose control and can't present a meaningful UI.

Reply:
> I was never entirely happy with this approach, and the recent "oops cameras don't work with pw 0.3" shows why: We treat pipewire as this magic black box. But it doesn't actually work, and we loose control and can't present a meaningful UI.

This is supposed to be a temporary solution until we can merge a real multimedia session manager into GNOME. It all ties together; security, policy, device configuration, the apps that are running, the user, ...

Right now we have an example session manager that works indeed like a black box and that tries to get info from various places (portal, permission store) to do its thing.

In any case, even if there is a non-black box GNOME controlled session manager, it will probably still need to access other parts of the system (config, permission store, ..) to do all the things it needs to do.